xernel/kernel/src/sched/thread.rs

```diff
@@ -167,3 +172,22 @@ impl Thread {
         self.process.upgrade()
     }
 }
+
+impl Drop for Thread {
+    fn drop(&mut self) {
+        if self.is_kernel_thread() {
```

Comment:
Case if user thread?

Reply:
The user thread stack is stored as a mapping in the Vm and is freed that way